### PR TITLE
chat-setting-ui

### PR DESCRIPTION
**Pull request by yashkapoor8710**
Simulator Screen Shot - iPhone 11 - 2020-02-24 at 17 14 56

### CODE REVIEW

File: pages/ChatSetting/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    return (
+        <>
+        <ChatProfileInfo
+          userImageSource={{ uri: "https://recraftrelic.com/images/Recraft_relic_web_logo_icon.png" }}
+          userName="Manoj Singh Negi"
+          status="I didn't change, I just grew up. You should try it once. God is really creative , I mean ..just look at me. I didn't change, I just grew up. You should try it once. God is really creative , I mean ..just look at me."
+          onButtonPress={goToChatList}
+          editInfo={true}
+        />
+        <ChatSettingTheme updateTheme={updateTheme} changeTheme="Dark Theme" logout="Logout" />
```
Comment (manojsinghnegiwd) [inline pages/ChatSetting/index.tsx]: Change name to `ThemeSetting`

File: pages/ChatSetting/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+interface Props extends RouteComponentProps {
+    dispatch: Dispatch
+  }  
+
+const ChatSetting: React.FunctionComponent<Props> = ({
+    dispatch,
+    history
+}: Props) => {
+      
+    const goToChatList = () => {
```
Comment (manojsinghnegiwd) [inline pages/ChatSetting/index.tsx]: change name to `goBack`

File: pages/ChatSetting/index.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import { setThemeAction } from '../../store/reducers/config';
+import { Dispatch } from 'redux';
+import ChatProfileInfo from '../../components/Chat/ChatProfileInfo';
+import ChatSettingTheme from '../../components/Chat/ChatSettingTheme';
+import ChatVersion from '../../components/Chat/ChatVersion';
+import { ThemeKey } from '../../config/themes';
+
+interface Props extends RouteComponentProps {
+    dispatch: Dispatch
+  }
```
Comment (manojsinghnegiwd) [inline pages/ChatSetting/index.tsx]: Please remove extra space.

File: pages/ChatList/index.tsx [hunk 4]
```diff
@@ -22,11 +22,15 @@ const ChatList: React.FunctionComponent<Props> = ({
...
+  const goToChatSetting = () => {
+    history.push('/setting')
+  }
+
   const updateTheme = (theme: AppTheme) => dispatch(setThemeAction(theme))
 
   return (
     <>
-      <ChatTitleBar updateTheme={updateTheme} />
+      <ChatTitleBar userImageSource={{ uri: "https://recraftrelic.com/images/Recraft_relic_web_logo_icon.png" }} onSettingPress={goToChatSetting}/>
```
Comment (manojsinghnegiwd) [inline pages/ChatList/index.tsx]: change name to `AppLogo`

File: config/themes.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    backgroundColor: "#ffffff",
+    highlightColor: "green",
+    highlightTextColor: "#ffffff",
+    textColor: "#333",
+    lightTextColor: "#b3b3b3",
+    lightBottomColor: "#e6e6e6",
+    alternateMessageBackgroundColor: '#B0E0E6',
+    modalBackgroundColor: 'rgba(0,0,0,0.5)',
+    warningColor: '#FF0000',
+    versionColor: '#B0E0E6',
```
Comment (manojsinghnegiwd) [inline config/themes.ts]: remove `versionColor`

File: components/Chat/ChatVersion.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { View, ViewStyle, StyleSheet, Dimensions} from 'react-native';
+import ThemedText from '../UI/ThemedText';
+
+interface Props {
+  name: string;
+};
+
+const { height, width } = Dimensions.get('window')
```
Comment (manojsinghnegiwd) [inline components/Chat/ChatVersion.tsx]: Add types here.

File: config/themes.ts (hunk 5 above)
Comment (yashkapoor8710) [reply config/themes.ts]: I have removed it

File: pages/ChatSetting/index.tsx (hunk 3 above)
Comment (yashkapoor8710) [reply pages/ChatSetting/index.tsx]: I have removed this

File: pages/ChatSetting/index.tsx (hunk 2 above)
Comment (yashkapoor8710) [reply pages/ChatSetting/index.tsx]: i have changed it

File: pages/ChatSetting/index.tsx (hunk 1 above)
Comment (yashkapoor8710) [reply pages/ChatSetting/index.tsx]: I have changed the name

File: components/Chat/ChatVersion.tsx (hunk 6 above)
Comment (yashkapoor8710) [reply components/Chat/ChatVersion.tsx]: I don't know the exact type so i have used 'any' Type

File: pages/ChatList/index.tsx (hunk 4 above)
Comment (yashkapoor8710) [reply pages/ChatList/index.tsx]: I have changed the name

File: router/index.tsx [hunk 7]
```diff
@@ -27,6 +28,7 @@ const Router: React.FunctionComponent<Props> = ({
               <Route exact path="/" component={ChatList} />
               <Route exact path="/chat/" component={ChatDetails} />
               <Route exact path="/profile" component={ChatProfileDetail} />
+              <Route exact path="/setting" component={ChatSetting} />
```
Comment (manojsinghnegiwd) [inline router/index.tsx]: change name to `Settings`

File: pages/ChatSetting/index.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    return (
+        <>
+        <ChatProfileInfo
+          userImageSource={{ uri: "https://recraftrelic.com/images/Recraft_relic_web_logo_icon.png" }}
+          userName="Manoj Singh Negi"
+          status="Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book."
+          onButtonPress={goBack}
+          editInfo={true}
+        />
+        <ThemeSetting updateTheme={updateTheme} />
```
Comment (manojsinghnegiwd) [inline pages/ChatSetting/index.tsx]: Change name to `ThemeToggle` and extract logout button

File: pages/ChatList/index.tsx [hunk 9]
```diff
@@ -3,30 +3,29 @@ import React from 'react';
...
 
-  const updateTheme = (theme: AppTheme) => dispatch(setThemeAction(theme))
+  const goToChatSetting = () => {
+    history.push('/setting')
+  }
 
   return (
     <>
-      <ChatTitleBar updateTheme={updateTheme} />
+      <AppLogo userImageSource={{ uri: "https://recraftrelic.com/images/Recraft_relic_web_logo_icon.png" }} onSettingPress={goToChatSetting}/>
```
Comment (manojsinghnegiwd) [inline pages/ChatList/index.tsx]: * Move the AppLogo image to `constants`.
* change `userImageSource` to `appLogoSource`

File: components/Chat/ThemeSetting.tsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+          }} thumbColor={theme.textColor} value={isDarkTheme} onValueChange={toggleDarkTheme} />
+        </View>
+      </View>
+      <TouchableOpacity>
+        <View style={[style.container, {borderColor: theme.lightBottomColor}]}>
+          <View style={style.leftContainer}>
+            <ThemedText styleKey="textColor">Logout</ThemedText>
+          </View>
+          <View style={style.rightContainer}>
+            <MaterialIcon name="logout" size={30} color={theme.warningColor} />
```
Comment (manojsinghnegiwd) [inline components/Chat/ThemeSetting.tsx]: Move logout outside.

File: components/Chat/ChatTitleBar.tsx [hunk 11]
```diff
@@ -1,42 +1,39 @@
-import React, { useState, useEffect } from 'react';
-import { View, TouchableOpacity, ViewStyle, StyleSheet, Switch, TextStyle } from 'react-native';
+import React from 'react';
```
Comment (manojsinghnegiwd) [inline components/Chat/ChatTitleBar.tsx]: Remove `chatTitleBar`

File: router/index.tsx (hunk 7 above)
Comment (yashkapoor8710) [reply router/index.tsx]: I have changed the name to settings

File: pages/ChatList/index.tsx (hunk 9 above)
Comment (yashkapoor8710) [reply pages/ChatList/index.tsx]: I have moved the AppLogo image source to constants & also renamed the imagesource prop

File: components/Chat/ThemeSetting.tsx (hunk 10 above)
Comment (yashkapoor8710) [reply components/Chat/ThemeSetting.tsx]: I have moved it outside the theme component

File: components/Chat/ChatTitleBar.tsx (hunk 11 above)
Comment (yashkapoor8710) [reply components/Chat/ChatTitleBar.tsx]: I have removed this file

File: pages/ChatSetting/index.tsx (hunk 8 above)
Comment (yashkapoor8710) [reply pages/ChatSetting/index.tsx]: I have changed the themesetting name to themetoggle

**Pull request merged**